### ISSUE TITLE
failAsyncPlatform called under valid conditions

### ISSUE THREAD

**Issue by StarLard** [apple/swift-argument-parser#605]
`failAsyncPlatform` is always called for `AsyncParsableCommand` with `AsyncParsableCommand` subcommands running on platforms macOS 10.15, macCatalyst 13, iOS 13, tvOS 13, watchOS 6, or above with package requirements at or above the aforementioned versions. This is due to a faulty platform [check](https://github.com/apple/swift-argument-parser/blob/c8ed701b513cf5177118a175d85fbbbcd707ab41/Sources/ArgumentParser/Parsable%20Types/ParsableCommand.swift#L129) which asserts that the desired platforms are available and then throws an error when they _are_ available.

**ArgumentParser version:** 
`main` branch.
**Swift version:**
```
swift-driver version: 1.62.15 Apple Swift version 5.7.2 (swiftlang-5.7.2.135.5 clang-1400.0.29.51)
Target: arm64-apple-macosx13.0
```

### Checklist
- [x] If possible, I've reproduced the issue using the `main` branch of this package
- [x] I've searched for [existing GitHub issues](https://github.com/apple/swift-argument-parser/issues)

### Steps to Reproduce
1. Create an `AsyncParsableCommand` root command with at least one `AsyncParsableCommand` subcommand.
2. Specify a minimum platform version at or above macOS 10.15, macCatalyst 13, iOS 13, tvOS 13, or watchOS 6.
3. Attempt to run in a debug configuration

### Expected behavior
Command executes successfully

### Actual behavior
[failAsyncPlatform](https://github.com/apple/swift-argument-parser/blob/c8ed701b513cf5177118a175d85fbbbcd707ab41/Sources/ArgumentParser/Parsable%20Types/ParsableCommand.swift#L247) is called and execution stops due to a fatal error.

**Comment by StarLard** [comment]
It seems that what this logic is intended to do is check if the specified platforms are _unavailable_, but I'm not sure why such a check would be needed as `AsyncParsableCommand` itself [annotated](https://github.com/apple/swift-argument-parser/blob/c8ed701b513cf5177118a175d85fbbbcd707ab41/Sources/ArgumentParser/Parsable%20Types/AsyncParsableCommand.swift#L14C1-L14C71) with `@available(macOS 10.15, macCatalyst 13, iOS 13, tvOS 13, watchOS 6, *)` and attempting to conform to it on a platform where it isn't available should throw a compiler error.

**Comment by StarLard** [comment]
Thanks! Would be great to clarify this in the error message.
